src/pool/pool_disjoint.c

```diff
@@ -804,11 +821,21 @@ umf_result_t disjoint_pool_free(void *pool, void *ptr) {
 
     bucket_t *bucket = slab->bucket;
 
-    VALGRIND_DO_MEMPOOL_FREE(pool, ptr);
     utils_mutex_lock(&bucket->bucket_lock);
 
-    utils_annotate_memory_inaccessible(ptr, bucket->size);
-    bucket_free_chunk(bucket, ptr, slab, &to_pool);
+    // TODO valgrind
+    VALGRIND_DO_MEMPOOL_FREE(pool, ptr);
+
+    // Get the unaligned pointer
+    // NOTE: the base pointer slab->mem_ptr needn't to be aligned to bucket size
+    size_t chunk_idx =
+        (size_t)floor((double)((uintptr_t)ptr - (uintptr_t)slab->mem_ptr) /
+                      slab->bucket->size);
```

Comment:
I thought about `ctgmath` header here, but I think, there is no need to convert types and do a floor since integer division does the same job.
```c
size_t chunk_idx = (((uintptr_t)ptr - (uintptr_t)slab->mem_ptr) /
                    slab->bucket->size);
```

Reply:
done